Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/services/DirectoryInitializationService.java

```diff
@@ -128,6 +159,15 @@ public static boolean areDolphinDirectoriesReady()
         return directoryState == DirectoryInitializationState.DOLPHIN_DIRECTORIES_INITIALIZED;
     }
 
+    public static String getUserDirectory()
+    {
+        if (userPath == null && directoryState != CANT_FIND_EXTERNAL_STORAGE)
```

Comment:
In my opinion, this shouldn't just check that we know what the path to the external storage, it should also check that `initializeExternalStorage()` has finished running. (And the exception message should say "DirectoryInitializationService has to finish running first!" then, I suppose.) This is because there shouldn't be any reason to get the path without wanting to access the directory, and trying to access parts of the directory while it's still being initialized can lead to inconsistent problems.